src/nvim/option.c

```diff
@@ -1490,7 +1490,7 @@ do_set (
               new_value_alloced = true;
               if (newval == NULL) {
                 newval = empty_option;
-              } else if (!(options[opt_idx].flags | P_NO_DEF_EXP)) {
+              } else if (!(options[opt_idx].flags & P_NO_DEF_EXP)) {
```

Comment:
? This does not look like there is anything wrong. Except that fix is probably testable and it looks like the current code should break test like https://github.com/ZyX-I/neovim/blob/f743b1b3cc951c27adea402227380d5b27d466de/test/functional/options/defaults_spec.lua#L222, but for options which both have something to expand in default value and no `P_NO_DEF_EXP` flag. E.g. `&helpfile`: this combination is rather uncommon and I failed to find other option which need default value expansion.